src/Microsoft.Framework.Runtime/Servicing/Breadcrumbs.cs

```diff
@@ -27,46 +46,83 @@ public Breadcrumbs(string breadcrumbsFolder)
+                    _logType,
+                    breadcrumbsFolder);
             }
         }
 
         public static bool IsPackageServiceable(PackageInfo package)
         {
+            if (!_isEnabled)
+            {
+                return false;
+            }
+
             // TODO: Figure out what makes a package serviceable
             return false;
         }
 
-        public void CreateBreadcrumb(string packageId, SemanticVersion packageVersion)
+        public void AddBreadcrumb(string packageId, SemanticVersion packageVersion)
         {
-            if (string.IsNullOrWhiteSpace(_breadcrumbsFolder))
+            if (!_isEnabled)
             {
                 return;
             }
 
-            // Create both files for now until we get clear instructions about the format of the name
-            CreateBreadcrumbFile(packageId);
-            CreateBreadcrumbFile(packageId + "." + packageVersion);
+            _breadcrumbsToWrite.Add(new BreadcrumbInfo()
+            {
+                PackageId = packageId,
+                PackageVersion = packageVersion
+            });
         }
 
-        public void CreateRuntimeBreadcrumb()
+        public void WriteAllBreadcrumbs(bool background = false)
         {
-#if DNX451
-            var runtimeAssembly = typeof(Breadcrumbs).Assembly;
-#else
-            var runtimeAssembly = typeof(Breadcrumbs).GetTypeInfo().Assembly;
-#endif
+            if (!_isEnabled)
+            {
+                return;
+            }
 
-            var version = runtimeAssembly
-                ?.GetCustomAttribute<AssemblyInformationalVersionAttribute>()
-                ?.InformationalVersion;
+            if (background)
+            {
+                ThreadPool.QueueUserWorkItem(_ =>
+                {
+                    WriteAllBreadcrumbsInternal();
+                });
+            }
+            else
+            {
+                WriteAllBreadcrumbsInternal();
+            }
+        }
 
-            if (!string.IsNullOrWhiteSpace(version))
+        private void WriteAllBreadcrumbsInternal()
+        {
+            foreach (var breadcrumb in _breadcrumbsToWrite)
             {
-                var semanticVersion = new SemanticVersion(version);
+                CreateBreadcrumb(breadcrumb.PackageId, breadcrumb.PackageVersion);
+            }
 
-                CreateBreadcrumb(runtimeAssembly.GetName().Name, semanticVersion);
+            _breadcrumbsToWrite.Clear();
+        }
+
+        /// <summary>
+        /// Writes a breadcrumb on the disk 
+        /// </summary>
+        /// <param name="packageId">The ID of the package</param>
+        /// <param name="packageVersion">The version of the the package</param>
+        private void CreateBreadcrumb(string packageId, SemanticVersion packageVersion)
+        {
+            if (string.IsNullOrWhiteSpace(_breadcrumbsFolder))
+            {
+                return;
```

Comment:
@davidfowl "Where exactly does it noop?" Here. I could change it to use the `_isEnabled` property instead an really do nothing.